api/v1beta1/bgppeer_types.go

```diff
@@ -81,6 +82,10 @@ type BGPPeerSpec struct {
 	// +optional
 	Password string `json:"password,omitempty" yaml:"password,omitempty"`
 
+	// secretRef is name of the authentication secret for BGP Peer
+	// +optional
+	SecretRef *v1.LocalObjectReference `json:"secretRef,omitempty" yaml:"secretRef,omitempty"`
```

Comment:
I just found out that there's a corev1.SecretReference which is maybe better https://pkg.go.dev/k8s.io/api/core/v1#SecretReference

Reply:
Also, I'd add this in the v1beta2 version (which https://github.com/metallb/metallb/pull/1245 is adding)

Reply:
good find! done.